guest/RATIONALE.md

```diff
@@ -47,7 +47,38 @@ mean amplifying that overhead to a point where second or longer latency would
 be possible. For reasons like this, we dismissed Go 1.21 for TinyGo, and will
 revisit when Go 1.22 introduces [`//go:wasmexport`][3].
 
+## Why do we use nottinygc?
+
+Kubernetes Scheduler plugins need to access very large model data, specifically
+node and pod data. WebAssembly has a sandbox model, so the memory of a plugin
+is not the same as host scheduler process. This means the node and pod are not
+passed by reference. In fact they cannot be copied by value either. Instead,
+they must be marshaled to wire format and unmarshalled within the WebAssembly
```

Comment:
The alternative is a large host abi to operate on the model directly. I don't recommend it, just that `must be marshaled` is possibly missing this alternative with its own problems

Reply:
PTAL I added a section from the prior decisions not here yet!